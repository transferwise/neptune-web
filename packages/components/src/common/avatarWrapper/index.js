import AvatarWrapper from './AvatarWrapper';

export default AvatarWrapper;
