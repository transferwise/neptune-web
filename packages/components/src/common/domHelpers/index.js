export function isTouchDevice() {
  return !!window?.matchMedia('(pointer: coarse)')?.matches;
}

export {
  addClickClassToDocumentOnIos,
  removeClickClassFromDocumentOnIos,
} from './documentIosClick';
