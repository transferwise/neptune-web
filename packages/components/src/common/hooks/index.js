export { default as useHasIntersected } from './useHasIntersected';
export { default as useClientWidth } from './useClientWidth';
