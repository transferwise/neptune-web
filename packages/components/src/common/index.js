export { default as HistoryNavigator } from './historyNavigator';
export { addNoScrollBodyClass, removeNoScrollBodyClass } from './DOMOperations';
export { Theme } from './theme';
// Props
export { Breakpoint } from './propsValues/breakpoint';
export { Size } from './propsValues/size';
export { Type } from './propsValues/type';
export { DateMode } from './propsValues/dateMode';
export { MonthFormat } from './propsValues/monthFormat';
// Position and Placement are duplicated and needs to be aligned with a breaking change
export { Position } from './propsValues/position';
export { Layout } from './propsValues/layouts';
export { FormControlType } from './propsValues/formControlType';
export { Status } from './propsValues/status';
export { Sentiment } from './propsValues/sentiment';
export { ProfileType } from './propsValues/profileType';
export { Key } from './key';
