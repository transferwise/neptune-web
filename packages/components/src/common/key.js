export const Key = {
  ENTER: 'Enter',
  SPACE: [' ', 'Spacebar'],
};
