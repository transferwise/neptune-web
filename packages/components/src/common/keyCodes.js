// @DEPRECATED
export default {
  DOWN: 40,
  UP: 38,
  LEFT: 37,
  RIGHT: 39,
  ENTER: 13,
  ESCAPE: 27,
  TAB: 9,
  SPACE: 32,
  BACKSPACE: 8,
};
