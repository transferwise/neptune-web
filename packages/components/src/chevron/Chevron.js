import '@transferwise/neptune-css/dist/css/chevron.css';
import React from 'react';
import Types from 'prop-types';
import classNames from 'classnames';
import { ChevronUp } from '@transferwise/icons';
import { Size, Position } from '../common';

const CONTAINER_SIZE = {
  [Size.EXTRA_SMALL]: 10,
  [Size.SMALL]: 16,
  [Size.MEDIUM]: 24,
};

const ExtraSmallChevron = ({ className }) => (
  <span
    className={classNames('tw-icon', 'tw-icon-chevron', className)}
    role="presentation"
    aria-hidden
  >
    <svg width="10" height="10" viewBox="0 0 10 10" fill="currentColor" focusable="false">
      <path
        fillRule="evenodd"
        d="M4.9995 2.0865L0.0635004 6.7645L1.2745 7.9125L4.9995 4.3835L8.7245 7.9135L9.9355 6.7645L4.9995 2.0865Z"
      />
    </svg>
  </span>
);

ExtraSmallChevron.propTypes = {
  className: Types.string,
};

ExtraSmallChevron.defaultProps = {
  className: null,
};

const Chevron = ({ orientation, size, disabled, className }) => {
  const classNameValue = classNames(
    'tw-chevron',
    { 'chevron-color': !disabled },
    `${[orientation]}`.toLowerCase(),
    className,
  );

  const sizeValue = CONTAINER_SIZE[size];

  if (size === Size.EXTRA_SMALL) {
    return <ExtraSmallChevron className={classNameValue} />;
  }

  return <ChevronUp className={classNameValue} size={sizeValue} />;
};

Chevron.Orientation = Position;
Chevron.Size = Size;

Chevron.propTypes = {
  orientation: Types.oneOf([
    Chevron.Orientation.TOP,
    Chevron.Orientation.BOTTOM,
    Chevron.Orientation.LEFT,
    Chevron.Orientation.RIGHT,
  ]),
  size: Types.oneOf([Chevron.Size.EXTRA_SMALL, Chevron.Size.SMALL, Chevron.Size.MEDIUM]),
  disabled: Types.bool,
  className: Types.string,
};

Chevron.defaultProps = {
  orientation: Chevron.Orientation.BOTTOM,
  size: Chevron.Size.SMALL,
  disabled: false,
  className: null,
};

export default Chevron;
