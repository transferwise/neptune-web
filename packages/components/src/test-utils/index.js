import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Provider } from '..';
import { DEFAULT_LOCALE } from '../common/locale';
import en from '../../i18n/en.json';

/**
 * Custom `render` function which wraps passed elements in Provider component
 * Source: https://testing-library.com/docs/example-react-intl/#creating-a-custom-render-function
 */
function customRender(ui, { locale = DEFAULT_LOCALE, messages = en, ...renderOptions } = {}) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return <Provider i18n={{ locale, messages }}>{children}</Provider>;
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
export { userEvent };
export { customRender as render };
