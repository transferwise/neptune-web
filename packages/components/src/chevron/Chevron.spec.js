import React from 'react';
import '@testing-library/jest-dom';
import { render } from '../test-utils';
import Chevron from '.';
import { Size } from '../common';

const { Orientation } = Chevron;

describe('Chevron', () => {
  describe('test rotation', () => {
    it('top', () => {
      const { container } = render(<Chevron orientation={Orientation.TOP} />);

      expect(container.querySelector('.tw-icon-chevron-up.top')).not.toBe(null);
    });

    it('right', () => {
      const { container } = render(<Chevron orientation={Orientation.RIGHT} />);

      expect(container.querySelector('.tw-icon-chevron-up.right')).not.toBe(null);
    });

    it('left', () => {
      const { container } = render(<Chevron orientation={Orientation.LEFT} />);

      expect(container.querySelector('.tw-icon-chevron-up.left')).not.toBe(null);
    });

    it('down (default)', () => {
      const { container } = render(<Chevron />);

      expect(container.querySelector('.tw-icon-chevron-up.bottom')).not.toBe(null);
    });
  });

  describe('disabled / enabled states', () => {
    it('enabled (by default)', () => {
      const { container } = render(<Chevron />);

      expect(container.querySelector('.tw-icon-chevron-up')).toHaveClass('chevron-color');
    });

    it('disabled', () => {
      const { container } = render(<Chevron disabled />);

      expect(container.querySelector('.tw-icon-chevron-up')).not.toHaveClass('chevron-color');
    });
  });

  describe('custom xs icon', () => {
    it('renders', () => {
      const { container } = render(<Chevron size={Size.EXTRA_SMALL} />);
      expect(container).toMatchSnapshot();
    });
  });
});
