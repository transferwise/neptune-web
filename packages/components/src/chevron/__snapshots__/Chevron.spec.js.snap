// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chevron custom xs icon renders 1`] = `
<div>
  <span
    aria-hidden="true"
    class="tw-icon tw-icon-chevron tw-chevron chevron-color bottom"
    role="presentation"
  >
    <svg
      fill="currentColor"
      focusable="false"
      height="10"
      viewBox="0 0 10 10"
      width="10"
    >
      <path
        d="M4.9995 2.0865L0.0635004 6.7645L1.2745 7.9125L4.9995 4.3835L8.7245 7.9135L9.9355 6.7645L4.9995 2.0865Z"
        fill-rule="evenodd"
      />
    </svg>
  </span>
</div>
`;
